frontend/src/components/NotificationSettingsModal/NotifcationSettingsModal.tsx

```diff
@@ -1,22 +1,102 @@
 import { X } from "lucide-react";
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { useAuthContext } from "@/Context/AuthProvider";
+import { updateNotificationSettings } from "@/lib/api_client";
+import { getNotificationSettings } from "@/lib/api_client";
 
 type NotificationSettingsModalProps = {
   onClose: () => void;
 };
 
 export default function NotificationSettingsModal({ onClose }: NotificationSettingsModalProps) {
-  const [pushEnabled, setPushEnabled] = useState(true);
-  const [importantEnabled, setImportantEnabled] = useState(true);
-  const [generalEnabled, setGeneralEnabled] = useState(true);
-  const [connectRequestEnabled, setConnectRequestEnabled] = useState(true);
-  const [connectionAcceptedEnabled, setConnectionAcceptedEnabled] = useState(true);
-  const [connectionDeniedEnabled, setConnectionDeniedEnabled] = useState(true);
-  const [privateEventInvitationEnabled, setPrivateEventInvitationEnabled] = useState(true);
-  const [eventRsvpUpdatesEnabled, setEventRsvpUpdatesEnabled] = useState(true);
-  const [publicEventAnnouncementsEnabled, setPublicEventAnnouncementsEnabled] = useState(true);
-  const [networkEventAnnouncementsEnabled, setNetworkEventAnnouncementsEnabled] = useState(true);
+  const { session } = useAuthContext();
+  const { setNotificationSettings } = useAuthContext();
+  const [pushEnabled, setPushEnabled] = useState<boolean | undefined>(undefined);
+  const [importantEnabled, setImportantEnabled] = useState<boolean | undefined>(undefined);
+  const [generalEnabled, setGeneralEnabled] = useState<boolean | undefined>(undefined);
+  const [connectRequestEnabled, setConnectRequestEnabled] = useState<boolean | undefined>(
+    undefined,
+  );
+  const [connectionAcceptedEnabled, setConnectionAcceptedEnabled] = useState<boolean | undefined>(
+    undefined,
+  );
+  const [connectionDeniedEnabled, setConnectionDeniedEnabled] = useState<boolean | undefined>(
+    undefined,
+  );
+  const [privateEventInvitationEnabled, setPrivateEventInvitationEnabled] = useState<
+    boolean | undefined
+  >(undefined);
+  const [eventRsvpUpdatesEnabled, setEventRsvpUpdatesEnabled] = useState<boolean | undefined>(
+    undefined,
+  );
+  const [publicEventAnnouncementsEnabled, setPublicEventAnnouncementsEnabled] = useState<
+    boolean | undefined
+  >(undefined);
+  const [networkEventAnnouncementsEnabled, setNetworkEventAnnouncementsEnabled] = useState<
+    boolean | undefined
+  >(undefined);
+
+  const [loading, setLoading] = useState(true);
+
+  async function handleSave() {
+    if (!session?.access_token) return;
 
+    try {
+      setNotificationSettings({
```

Comment:
As before, ideally you should update on the server and then refetch to keep the data synced with it